tests/Functional/ContentRenderTestBase.php

```diff
@@ -97,4 +98,110 @@ protected function assertMediaDocumentDefaultRender(NodeElement $element, string
     $this->assertContains('Download', $file_download_link->getText());
   }
 
+  /**
+   * Creates media image entity.
+   *
+   * @param string $name
+   *   Name of the image media.
+   *
+   * @return \Drupal\media\MediaInterface
+   *   Media image instance.
+   */
+  protected function createMediaImage(string $name): MediaInterface {
+    // Create file instance.
+    $file = file_save_data(file_get_contents(drupal_get_path('theme', 'oe_theme') . '/tests/fixtures/placeholder.png'), "public://placeholder_$name.png");
+    $file->setPermanent();
+    $file->save();
+
+    $media = $this->getStorage('media')->create([
+      'bundle' => 'image',
+      'name' => "Test image $name",
+      'oe_media_image' => [
+        'target_id' => (int) $file->id(),
+        'alt' => "Alternative text $name",
+      ],
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $media->save();
+
+    return $media;
+  }
+
+  /**
+   * Asserts featured media field rendering.
+   *
+   * @param \Behat\Mink\Element\NodeElement $rendered_element
+   *   Rendered element.
+   * @param string $name
+   *   Name of the image media.
+   */
+  protected function assertFeaturedMediaField(NodeElement $rendered_element, string $name): void {
+    $figures = $rendered_element->findAll('css', 'figure.ecl-media-container');
+    $this->assertCount(1, $figures);
```

Comment:
Reset the array.
```suggestion
    $this->assertCount(1, $figures);
    $figures = reset($figures);
```

Reply:
Updated when the array is used after assertCount

Reply:
The same as above. Just reseting the array but not saving the element doesn't help much.